modules/partition-replicator/src/main/java/org/apache/ignite/internal/partition/replicator/PartitionReplicaLifecycleManager.java

```diff
@@ -969,7 +989,10 @@ private CompletableFuture<Void> stopAndDestroyPartitionAndUpdateClients(
             Set<Assignment> stableAssignments,
             Assignments pendingAssignments,
             boolean isRecovery,
-            long revision
+            long revision,
+            // TODO: Remove after https://issues.apache.org/jira/browse/IGNITE-24374
+            // It's a workaround because with TestPlacementDriver we couldn't reserve primary replicas that's we we shouldn't stop them now.
```

Comment:
```suggestion
            // It's a workaround because with TestPlacementDriver we can't reserve primary replicas so we shouldn't stop them now.
```